domains/pallets/messenger/src/lib.rs

```diff
@@ -48,6 +48,14 @@ use sp_messenger::messages::{
 use sp_runtime::traits::{Extrinsic, Hash};
 use sp_runtime::DispatchError;
 
+/// Maximum number of XDMs per domain/channel with future nonces that are allowed to be validated.
+/// Any XDM comes with a nonce above Maximum future nonce will be rejected.
+const MAX_FUTURE_ALLOWED_NONCES: u32 = 20;
+
+/// Transaction validity for a given validated XDM extrinsic.
+/// If the extrinsic is not included in the bundle, extrinsic is removed from the TxPool.
+const XDM_TRANSACTION_LONGEVITY: u64 = 5;
```

Comment:
This also seems a bit small to me, as even if the XDM is included by the bundle, the bundle itself doesn't guarantee included by the next consensus block thus the next domain block, since we just try to above the XDM stick in the tx pool forever IMO it is okay to set this value to something like 10 or 20.

Reply:
hmm i dont understand your reasoning. Longevity is related to domain blocks and current 5 means, xdm can be valid for the next 5 domain blocks